docs/pages/docs/core-concepts/filtering.mdx

```diff
@@ -8,55 +8,67 @@ import HeartIcon from "@heroicons/react/solid/HeartIcon";
-be in the final entrypoint selection.
+- Filter by [workspace name](#filter-by-workspace-name)
+- Filter by [workspace directory](#filter-by-directory)
+- Include [dependents](#include-dependents-of-matched-workspaces) and [dependencies](#include-dependencies-of-matched-workspaces) of matched workspaces
+- Execute tasks from the [workspace root](#the-workspace-root)
+- Filter by [changes in git history](#filter-by-changed-workspaces)
+- [Exclude workspaces](#excluding-workspaces) from selection
 
-Turborepo's filter syntax is based on pnpm's filter syntax, so it should feel familiar to pnpm users.
+Turborepo will run each task against each matched workspace, ensuring that any dependent tasks are run first, according to the `pipeline` specification in [`turbo.json`](/docs/reference/configuration#pipeline).
 
-### Filter by workspace
+## Filter Syntax
 
-Supports exact matches (`--filter=my-pkg`), and globs (`--filter=*my-pkg*`) are supported.
-Scoped workspaces (`@foo/bar`) can be abbreviated to just the workspace name (`bar`) as long as the workspace name
-is unique within the monorepo (e.g. `@foo/bar` exists, but `@baz/bar` does not).
+### Multiple filters
 
-The monorepo's root can be selected using the token `//`.
+You can specify multiple filters by adding multiple `--filter` to the command:
+
+```sh
+turbo run build --filter=my-pkg --filter=my-app
+```
+
+### Filter by workspace name
+
+Supports exact matches (`--filter=my-pkg`), and globs (`--filter=*my-pkg*`).
 
 ```sh
 # Build 'my-pkg', letting `turbo` infer task dependencies
 # from the pipeline defined in turbo.json
 turbo run build --filter=my-pkg
 
-# Build '@foo/bar', letting `turbo` infer task dependencies
+# Build '@acme/bar', letting `turbo` infer task dependencies
 # from the pipeline defined in turbo.json
-turbo run build --filter=@foo/bar
+turbo run build --filter=@acme/bar
 
 # Build all workspaces that start with 'admin-', letting turbo infer task
 # dependencies from the pipeline defined in turbo.json
 turbo run build --filter=admin-*
+```
 
-# Run the format script from the root "package.json" file:
-turbo run format --filter=//
+#### Scopes
+
+Some monorepos prepend their workspace names with a scope, such as `@acme/ui` and `@acme/app`. As long as the scope (`@acme`) is unique across the codebase, you may omit it from filters.
+
+```diff
+- turbo run build --filter=@acme/ui
++ turbo run build --filter=ui
 ```
 
 ### Include dependents of matched workspaces
 
-Prepend `...` to the filter. If `my-app` depends on `my-lib`, `...my-lib` will select `my-app` and `my-lib`.
-Optionally including a `^` (`...^my-lib`) will select all of `my-lib`'s dependents, but not `my-lib` itself. In this case, just `my-app` will be
-selected.
+If `my-app` depends on `my-lib`, `...my-lib` will select `my-app` and `my-lib`.
+
+Including a `^` (`...^my-lib`) will select all of `my-lib`'s dependents, but not `my-lib` itself.
```

Comment:
I'm not sure _where_ to do this, but when I read these docs, it would have been helpful to have example use cases of all these. E.g. in this case, something like: "In an app + shared packages" situation, this can be helpful if you want to build all of the dependencies of an app in the monorepo, but not the app itself".  (or whatever is an appropriate example). 

I got a bit overwhelmed learning all the possibilities for filtering, but not knowing how/when to apply them.

Reply:
Gotcha, good point!